src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.cs

```diff
@@ -98,9 +96,10 @@ public Rfc2898DeriveBytes(string password, int saltSize, int iterations, HashAlg
             RandomNumberGenerator.Fill(_salt.AsSpan(0, saltSize));
 
             _iterations = (uint)iterations;
-            _password = Encoding.UTF8.GetBytes(password);
+            byte[] passwordBytes = Encoding.UTF8.GetBytes(password);
             HashAlgorithm = hashAlgorithm;
-            _hmac = OpenHmac();
+            _hmac = OpenHmac(passwordBytes);
+            CryptographicOperations.ZeroMemory(passwordBytes);
```

Comment:
This behaviour is inconsitent with the `public Rfc2898DeriveBytes(string password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm)` constructor, which also does GetBytes, but doesn't zero them afterwards.

Reply:
@bartonjs @GrabYourPitchforks What do you think? Shuffle the constructor logic around so we can get this one, too? https://github.com/dotnet/runtime/pull/47483/files#diff-08325fdc91c756ac12cd9ac088a682a266578fe47620cb8b4604d6883370405fR72

Reply:
I went ahead and changed it; happy to go down a different path.

Reply:
If you're bored, a reasonable followup would be to collapse the two remaining ctors into one.  Assuming my quick read was correct, it'd be something like

```C#
private Rfc2898DeriveBytes(byte[] password, byte[]? salt, int saltLength, int iterations, HashAlgorithmName hashAlgorithm, bool clearPassword)
{
// salt.Length => saltLength
// if salt is null, RNG fill, else copy.
// otherwise everything else is the same?
}

public Rfc2898DeriveBytes(string password, int saltSize, int iterations, HashAlgorithmName hashAlgorithm)
    : this(Encoding.UTF8.GetBytes(password), null, saltSize, iterations, hashAlgorithm, clearPassword: true) { }

public Rfc2898DeriveBytes(string password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm)
    : this(Encoding.UTF8.GetBytes(password), salt, salt?.Length ?? 0 : iterations, hashAlgorithm, clearPassword: true)

etc as necessary.

```